Tests/UtilityTests/ProcessTests.swift

```diff
@@ -100,10 +100,43 @@ class ProcessTests: XCTestCase {
             XCTAssertFalse(try Process.running(ProcessID(child), orDefunct: true))
         }
     }
+    
+    func testThreadSafetyOnWaitUntilExit() throws {
+        let process = Process(args: "echo", "hello")
+        try process.launch()
+        
+        var result1: String = ""
+        var result2: String = ""
+        
+        let t1 = Thread {
+            do {
+                result1 = try process.waitUntilExit().utf8Output()
+            } catch {
+                XCTFail("Process did not exit gracefully")
+            }
+        }
+        
+        let t2 = Thread {
+            do {
+                result2 = try process.waitUntilExit().utf8Output()
+            } catch {
+                XCTFail("Process did not exit gracefully")
```

Comment:
The assertion text is not exactly correct, the method throws if the output isn't valid utf8.
We can use try! here